src/lib/components/table/table-column-settings/table-column-settings.component.spec.ts

```diff
@@ -0,0 +1,66 @@
+import {
+    ComponentFixture,
+    TestBed
+} from '@angular/core/testing';
+import { LocalizationModule } from 'angular-l10n';
+import { l10nConfig } from '../../../../app/translation/l10n.config';
+import { TableColumnSettingsComponent } from './table-column-settings.component';
+import Spy = jasmine.Spy;
+import {
+    TerraButtonComponent,
+    TooltipDirective
+} from '../../..';
+import {
+    MatDialog,
+    MatDialogModule
+} from '@angular/material/dialog';
+import { Overlay } from '@angular/cdk/overlay';
+import { Router } from '@angular/router';
+import { MockRouter } from '../../../testing/mock-router';
+import { TableColumnSettingsDialogComponent } from './dialog/table-column-settings-dialog.component';
+import { MatListModule } from '@angular/material/list';
+import { FormsModule } from '@angular/forms';
+
+describe('TableColumnSettingsComponent', () =>
+{
+    let component:TableColumnSettingsComponent;
+    let fixture:ComponentFixture<TableColumnSettingsComponent>;
+
+    const router:MockRouter = new MockRouter();
+
+    beforeEach(() =>
+    {
+        TestBed.configureTestingModule(
+            {
+                declarations: [TableColumnSettingsComponent,
+                                TableColumnSettingsDialogComponent,
+                               TooltipDirective,
+                               TerraButtonComponent],
+                imports:      [
+                    LocalizationModule.forRoot(l10nConfig),
+                    MatDialogModule,
+                    MatListModule,
```

Comment:
Ist für diesen Test eigentlich nicht notwendig, oder?

Reply:
die dialogref in der component wird irgendwie nicht richtig erzeugt, ich dachte es liegt daran dass die dialogComponent nicht declared wird und habe sie deswegen hinzugefügt mit den nötigen imports dazu
das löst das problem aber leider auch nicht